src/utils/OrderQuoter.ts

```diff
@@ -29,15 +29,21 @@ export enum OrderValidation {
   OK,
 }
 
-export interface ResolvedOrder {
+export type ResolvedOrder = ResolvedDutchOrder | ResolvedRelayOrder;
```

Comment:
yea I think this is going to be very disruptive for importers, doing `instanceof` de-unioning is very jank in ts for library importers

Reply:
I think this is clear indication that we are over-using inheritance here between RelayOrder vs X orders. My suggestion to allow some abstraction over them (i.e. abstract over offchain orders while not trying to abstract over Relay vs X) is something like:

define an _interface_ that describes functions all _offchain_ orders should have. Probably includes `hash()`, `permitData()`, `serialize()` etc.

`Order` type => `UniswapXOrder`, and have it implement your new interface
`RelayOrder` type standalone, also implements the interface

Reply:
Ah yes this makes sense to me. I was struggling with the existing inheritance model in the repo but couldn't figure out why - this helps a lot. I think defining that interface and having each individual class implement it is way cleaner